misc/mdns/src/behaviour.rs

```diff
@@ -37,28 +37,46 @@ use wasm_timer::{Delay, Instant};
 /// them to the topology.
 pub struct Mdns<TSubstream> {
     /// The inner service.
-    service: MdnsService,
+    service: MaybeBusyMdnsService,
 
     /// List of nodes that we have discovered, the address, and when their TTL expires.
     ///
     /// Each combination of `PeerId` and `Multiaddr` can only appear once, but the same `PeerId`
     /// can appear multiple times.
+    //
+    // TODO: Why this optimization? Is mdns ever within the hot path of an
+    // application? Is the access latency on this data structure relevant?
     discovered_nodes: SmallVec<[(PeerId, Multiaddr, Instant); 8]>,
 
-    /// Future that fires when the TTL at least one node in `discovered_nodes` expires.
+    /// Future that fires when the TTL of at least one node in `discovered_nodes` expires.
     ///
     /// `None` if `discovered_nodes` is empty.
+    // TODO: Would a simple std Instant suffice as well?
```

Comment:
We currently report expired nodes as events of the behaviour. If we want these to be reported "pro-actively", then it has to be a `Delay`.

Reply:
Makes sense. Thanks!